x/eibc/keeper/keeper.go

```diff
@@ -177,6 +178,11 @@ func (k Keeper) ListDemandOrdersByStatus(ctx sdk.Context, status commontypes.Sta
 	for ; iterator.Valid(); iterator.Next() {
 		var val types.DemandOrder
 		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		for _, opt := range opts {
```

Comment:
the `limit(limit int)` option is bit miss the point here no?
u'll still process and unmarshal the entire query
I think better design for limit should break this iterator